Library/Formula/open-mpi.rb

```diff
@@ -27,7 +29,7 @@ class OpenMpi < Formula
   conflicts_with "lcdf-typetools", :because => "both install same set of binaries."
 
   depends_on :java => :build
-  depends_on :fortran => :recommended
+  depends_on :fortran if build.with?("fortran")
```

Comment:
I completely agree! As I said in the other PR, this is just a semi-ugly workaround and not the real solution. I myself would like to see all of those conditional `depends_on` eliminated (those within an `if`/`unless` block or with a trailing `if`/`unless` modifier). But fixing the issue here and eradicating conditional `depends_on` requires an in-depth understanding of Homebrew's dependency resolution logic – something I don't have (yet).